### DISCUSSION TITLE
Stiching multiple sequences together

### DISCUSSION THREAD

**Discussion by Wheart** [time4tea/gopro-dashboard-overlay#175] [Q&A]
Hi!
At first: thanks for the excellent software - it really does it's work!
I came to the problem: gopro-join.py stitches together series (like GH010095.MP4, GH020095.MP4, GH030095.MP4).
But usually there are more series from a single trip and sometimes I use time lapse mode on "boring" sections.
Is it possible to stitch together multiple series? 
Is it possible to stitch together normal and time lapse videos and then generate overlay (with added external GPX data)?

Regards,
Wojciech Penar

**Comment by bphilli7** [comment]
Hello, I successfully stitched together multiple sequences of video into a single video with an overlay and data from a gopro hero 10 (GPS data included) and a fit tracker device (Wahoo Elemnt Bolt, FIT file). Here is the process I used.

1) Rename individual GoPro videos to be captured and joined by gopro-join.py, if they were taken with individual key presses and not auto-split by the GoPro. For example, a sequence of three files (GH010095.MP4, GH010096.MP4, GH010097.MP4) needs to be named "GH010095.MP4, GH020095.MP4, GH030095.MP4".
2) Use gopro-join.py to join the sequence together into a single, joined output video.
3) Run gopro-dashboard.py on the joined output video with all your preferred settings/layout options
4) Optional: merge an external GPX/FIT source into the overlay data: use the options "--gpx-merge OVERWRITE" and --fit/gpx <filename>

gopro-dashboard.py will then make a single video with an overlay of the all the data and where there are "gaps" in time between videos the merged sequence will jump to the correct data in the gpx/fit data set. It worked very smoothly and I'm very happy with the result.

Warning:
Do not use --use-gpx-only or --use-fit-only. These will cause an overlay that will be out of sync with the joined video, as the dashboard script seems to match video frames to overlay data based on where they each fall in time after the start of the recording. So for a stitched file of multiple video sequences, the gaps which are recorded by the GPX/FIT tracker will be used as overlay data for video shots taken after the gap in time, which will be incorrect. I would like to see the dashboard use the GoPro timestamp and the gpx/fit tracker timestamp to align correct gpx/fit data to the video frames, but I don't think it works this way now. Another corner case would be gaps in the gpx/fit tracker dataset for which video is still rolling. In this case it would be nice to either hold the previous overlay data or set all the metrics to their null value of '-', until the gpx/fit tracker begins collecting data again.

Another warning:
GoPro "Timewarp" has an "Auto" timescale option which dynamically scales the framerate. I have tried to use dashboard on timewarp videos with this Auto timescale option set and it resulted in video/overlay data wildly out of sync. Lots of GPS data errors were thrown by gopro-dashboard.py. I am hoping to do some more experiments with Timewarp set to a fixed timescale option and stitch those in to the sequence following my process from above. Hopefully they stay in sync with the overlay data.
